scripts/firedrake-install

```diff
@@ -305,8 +319,122 @@ def pip_requirements(package):
         stdout.write("No dependencies found. Skipping.\n")
 
 
+if travis:
+    # Only enabled in travis mode
+    # Utilise travis directory caching for petsc/petsc4py dependencies
+    class directory(object):
```

Comment:
I think the convention for classes that define context managers is that they are lower-case.